src/modules/jcms_rest/src/EventSubscriber/ExceptionJcmsJsonSubscriber.php

```diff
@@ -16,4 +19,90 @@ protected function getHandledFormats() {
     return ['jcms_json'];
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  protected static function getPriority() {
+    // This will fire after the most common HTML handler, since HTML requests
+    // are still more common than JSON requests.
+    return -75;
+  }
+
+  /**
+   * Handles a 400 error for JCMS JSON.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent $event
+   *   The event to process.
+   */
+  public function on400(GetResponseForExceptionEvent $event) {
+    $response = new JsonResponse(array('message' => $event->getException()->getMessage()), Response::HTTP_BAD_REQUEST, $event->getException()->getHeaders());
+    $event->setResponse($response);
+  }
+
+  /**
+   * Handles a 403 error for JCMS JSON.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent $event
+   *   The event to process.
+   */
+  public function on403(GetResponseForExceptionEvent $event) {
+    $response = new JsonResponse(array('message' => $event->getException()->getMessage()), Response::HTTP_FORBIDDEN, $event->getException()->getHeaders());
+    $event->setResponse($response);
+  }
+
+  /**
+   * Handles a 404 error for JCMS JSON.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent $event
+   *   The event to process.
+   */
+  public function on404(GetResponseForExceptionEvent $event) {
+    $response = new JsonResponse(['title' => $event->getException()->getMessage()], Response::HTTP_NOT_FOUND, $event->getException()->getHeaders());
+    $event->setResponse($response);
+  }
+
+  /**
+   * Handles a 405 error for JCMS JSON.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent $event
+   *   The event to process.
+   */
+  public function on405(GetResponseForExceptionEvent $event) {
+    $response = new JsonResponse(array('title' => $event->getException()->getMessage()), Response::HTTP_METHOD_NOT_ALLOWED, $event->getException()->getHeaders());
```

Comment:
Seems to conform to https://github.com/elifesciences/api-raml/blob/develop/src/model/error.v1.yaml :+1: